app/scripts/migrations/097.ts

```diff
@@ -0,0 +1,49 @@
+import { cloneDeep, isEmpty } from 'lodash';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+export const version = 97;
+
+/**
+ * Remove `nonceDetail` from transactions
+ *
+ * @param originalVersionedData
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(state: Record<string, any>) {
+  const TransactionController = state?.TransactionController || {};
+  const transactions = state?.TransactionController?.transactions || {};
+
+  if (isEmpty(TransactionController) || isEmpty(transactions)) {
+    return;
+  }
+
+  const newTxs = Object.keys(transactions).reduce((txs, txId) => {
+    const transaction = transactions[txId];
+    if (transaction?.nonceDetails) {
+      delete transaction.nonceDetails;
+    }
+    return {
+      ...txs,
+      [txId]: transaction,
+    };
+  }, {});
+
+  state.TransactionController = {
+    ...TransactionController,
+    transactions: {
```

Comment:
Minor, but can we just assign here rather than using a spread operator?

Reply:
Done

Reply:
Done